### PR TITLE
[DebugInfo] Get rid of redundant conditional checks in `/DebugInfo`

### LINKED ISSUE #90326: llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h: 3 * redundant condition ?
Static analyser cppcheck says:

llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h:249:29: style: Redundant condition: Parent.Element. '!A || (A && B)' is equivalent to '!A || B' [redundantCondition]

Source code is

    assert((!Parent.Element ||
            (Parent.Element && static_cast<LVElement *>(Parent.Element))) &&
           "Invalid element");

Duplicates with the scope and the symbol a few lines later.

### PR DESCRIPTION
**Pull request by Rajveer100**
Resolves #90326

### CODE REVIEW

**Comment by Rajveer100** [comment]
cc @JOE1994

**Review by JOE1994 — APPROVED** [review]

**Comment by Rajveer100** [comment]
Thanks for the approval! Could you land this for me?

File: llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h [hunk 1]
```diff
@@ -246,20 +246,17 @@ class LVObject {
   virtual void setName(StringRef ObjectName) {}
 
   LVElement *getParent() const {
-    assert((!Parent.Element ||
-            (Parent.Element && static_cast<LVElement *>(Parent.Element))) &&
+    assert((!Parent.Element || static_cast<LVElement *>(Parent.Element)) &&
```
Comment (JOE1994) [inline llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h:249]: Upon another look, the original assert (before this mod) looks flawed.
`(! Parent.Element)` always evaluates to `true` except for when `Parent.Element` is an all-1 bitvector..
Could you update this and the other 2 assert conditions like below?

```suggestion
    assert(((Parent.Element == nullptr) || static_cast<LVElement *>(Parent.Element)) &&
```

**Review by JOE1994 — CHANGES_REQUESTED** [review]

File: llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h (hunk 1 above)
Comment (JOE1994) [reply llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h:249]: Oops, please disregard my previous comment.. `Parent.Element` is converted to bool before the `!` operator is applied, so there aren't any issues with the original assert.

**Review by JOE1994 — APPROVED** [review]

File: llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h (hunk 1 above)
Comment (Rajveer100) [reply llvm/include/llvm/DebugInfo/LogicalView/Core/LVObject.h:249]: No problem!

**Pull request merged**